### ISSUE TITLE
Регистрация

### ISSUE THREAD

**Issue by oDenZeL** [tym32167/arma3beclient#49]
В принципе, если бы был реализован инсталятор, можно было бы сделать регистрацию при первом запуске программы.
Допустим нужно ввести - Никнейм, PID, и @.mail к примеру.

Опять же, синхронизовав с личной базой данных, можно было бы увидеть кто кого заблокировал.
Во вкладке "Администрация" корректо бы отображались никнеймы и другая информация.

**Comment by tym32167** [comment]
> регистрацию при первом запуске программы.

Регистрацию где? У программы нет своего сервера.

> Никнейм, PID, и @.mail к

Что такое PID?

> можно было бы увидеть кто кого заблокировал

Только если все админы пользуются одной программой. Да и зачем это, если админ, что забанил, пишется в комментарии к бану?

> Во вкладке "Администрация" корректо бы отображались никнеймы и другая информация

У нас же нет такой вкладки. Никнеймы кого отображались бы корректно?

Если честно, я давно подумываю о PTP синхронизации, когда, например, два и более админов установят у себя проги, расшарят порты и смогут соединить свои проги в синхронизированную сеть. Это, правда, будет работать тлько при статическом IP или динамическом DNS. Зато это легко конвертируется в одну прогу-центральную и несколько синхронизированных нод. Однако, такое дело требует отдельного релиза и я хз когда до этого дойду. 

Есть идея гораздо круче - написать сервер с подобным, как в проге, функционалом и веб-интерфейсом, чтобы поддерживало множество пользователей, ролей, серверов и тд. Чтобы зайти можно было в интерфейс хоть с компьютера, хоть с планшета, хоть с телефона. Но пока к этой задаче нет времени подступиться, как только время появится - сделаю концепт, и там поглядим.

**Comment by ToxaBes2** [comment]
Чисто под веб есть вот такое: https://github.com/Nizarii/arma-rcon-class-php
php сокетами по udp можно работать с battleye сервером, легко дописывается. Сможет ли покрыть весь функционал - хз.

**Comment by tym32167** [comment]
Так это, как я понял, просто прокси к серверу.  Там ничего нет, ни
интерфейса, ни базы данных.

**Comment by ToxaBes2** [comment]
Это механизм выполнения команд battleye сервера и получения результата. Если ты используешь только команды battleye+ steam query то с его помощью можно на любом современном php фреймворке накидать админку которая будет повторять функционал программы и хранить данные в бд, например MySQL. Это вариант для кросс-платформенного решения, чисто под веб, поставил на веб-сервер и работает. Единая точка точка входа для всех админов, единая база и т.д.

Я так понял ты задумываешься над написанием сервера с веб частью который будет ставиться только под Win (ибо .NET)?

**Comment by tym32167** [comment]
Да, думаю про веб, но пока это только мысли :) Про платформы не думал еще,
но есть возможность сделать это на .net core, что должен запускаться и под
линуском.
Вся моя прога тоже типа админки, сам код для отправки получения сообщений
небольшой и я его взял с другого проекта.
Не думаю, что так легко будет накидать админку поверх этих пхп скриптов, по
крайней мере с тем же функционалом, что у нас.
А вообще если говорить про аналоги, я слышал про BEC и BigBrotherBot. Это
вроде тоже хостед решения, но у меня не было возможности их попробовать.

**Comment by ToxaBes2** [comment]
> что должен запускаться и под линуском.

Это было бы здорово (для меня). Универсальная прога которая работает под Win/Lin + может работать как несколько независимых клиентов (client version) - у каждого админа своя база, свой экземпляр проги и они мудохаются с синхронизацией, так и в режиме сервера (server version) - один установленный экземпляр на Win/Lin с доступом по веб-морде которая может открываться на любом указанном порту с единой базой и ACL для админов. При таком варианте все остальные аналоги просто перестают быть нужными, т.к. покрываются сразу все варианты использования.

Если решишься такое реализовать - готов быть твоим альфа-тестером серверного Linux варианта.

**Comment by ToxaBes2** [comment]
По серверному варианту: служба по Win и демон (systemd?) под Lin. После запуска открываем в браузере http://127.0.0.1:32167 и видим форму логина в админку. Как-то так.

**Comment by tym32167** [comment]
Понял, спасибо. Буду тебя иметь ввиду :)

**Comment by tym32167** [comment]
Я сам пока не очень представляю, как это будет работать под линуксом, но вот тут есть занимательная инфа https://www.microsoft.com/net/core

**Comment by tym32167** [comment]
По поводу самого сервера, мне кажется, на линуксе от него заведется только сервер+веб часть. Клиентскую прогу для коннекта к серверу писать планов нет. Собственно, пока с планами вцелом не очень. Мне бы текущий релиз выпустить, а за следующий я, наверное, коли возьмусь, так в след году.

**Comment by ToxaBes2** [comment]
> а в линуксе от него заведется только сервер+веб часть. 

Интересна серверная часть с веб мордой, в этом случае без клиента можно обойтись.

> за следующий я, наверное, коли возьмусь, так в след году

У тебя просто слишком много пивных на кв км ) 
Время не критично на самом деле.

**Comment by tym32167** [comment]
@ToxaBes после такого https://www.battlemetrics.com/home даже сомневаюсь, браться вообще за веб или нет O_o

**Comment by ToxaBes2** [comment]
Это просто стата, через них не забанишь человека, не разбанишь его, не напишешь в чат в игру и тд.
Запилить статистику не проблема -> http://mon.tehgam.com/?s=1 
А вот реализовать управление, это совсем другое дело.

**Comment by tym32167** [comment]
@ToxaBes они за 5 баксов предлагают RCON доступ через веб
https://www.battlemetrics.com/rcon/about

**Comment by ToxaBes2** [comment]
В данном конкретном случае я не готов платить 5 баксов в месяц за то что могу сам накидать на php+bootstrap за несколько дней (основной функционал). Также мне не нравится вариант в котором данные лежат на левом сервере.
Делать или нет решать тебе. Я предлагал web часть как вариант кросплатформенности, т.к. меня интересует Linux-версия которую можно было бы поставить на собственный сервер.

**Comment by vosur** [comment]
> Во вкладке "Администрация" корректо бы отображались никнеймы и другая информация

Я так понимаю, имеется ввиду отображение в вкладке Admins хотя бы поля "Ник", который будет брать ip адрес и выдавать ник последнего пользователя, который его использовал. Так будет легче определять, у кого запущена данная программа

**Comment by tym32167** [comment]
@vosur  чтобы показывать ник админа, надо знать, какой админ на каком IP сидит (при этом, в теории, с одного IP благополучно могут сидеть несколько админов). Просто запущенная у клиента прога не имеет ни малейшего понятия о запущенных других прогах и их админах, потому это в текущей проге просто не реализуемо.

**Comment by ToxaBes2** [comment]
Для этого нужна единая точка входа (веб-морда). 
Каждый админ использует свой логин/пароль для входа и после авторизации можно будет внизу показывать строку вида:
`Админы онлайн: ..., ..., ... .`
Примерно как на форумах пишется.

**Comment by tym32167** [comment]
@ToxaBes  веб версия уже в работе, там подобное будет (при условии что админы через веб версию заходят, а не через прогу). MVP планирую к концу года выкатить, стек будет MySQL + asp.net core 2.1 на сервере и React+Redux на клиенте (будет работать на Win + Linux + Mac на сервере и все современные браузеры на клиенте). Прототип уже сейчас мониторит сервера.

**Comment by ToxaBes2** [comment]
Ок, у .NET Core 2.1 в списке поддерживаемых ОС есть openSUSE 42.3 x64
У меня на сервере 42.2, думаю заведется. Как будет что тестить - зови.

**Comment by tym32167** [comment]
@ToxaBes  договорились
